clients/src/main/java/org/apache/kafka/common/utils/ChunkedBytesStream.java

```diff
@@ -291,7 +291,7 @@ public long skip(long toSkip) throws IOException {
         // Skip bytes stored in intermediate buffer first
         int avail = count - pos;
         long bytesSkipped = (avail < remaining) ? avail : remaining;
-        pos += bytesSkipped;
+        pos += (int) bytesSkipped;
```

Comment:
This is the main blocker remaining before this PR can be non draft.

Reply:
@ijuma casting to int here is safe. Can you please add the following comment here:
```
// cast to int is fine here since the value of bytesSkipped is upper bound by value of avail
// which is an int
```

It is safe because as per the lines above it `long bytesSkipped = (avail < remaining) ? avail : remaining;`, bytesSkipped is minimum of avail or remaining which means it's max value is available which we know is an int.

Same for the other place.

Reply:
Hmm, in that case we can perhaps change `bytesSkipped` to be an `int`?

Reply:
I am afraid that won't be possible.

This is because there are two branches where `bytesSkipped` is updated.  In one branch when we call   `bytesSkipped = getInIfOpen().skip(remaining)`,  `bytesSkipped` holds a long value since `getInIfOpen().skip(remaining)` returns a long. In the other branch, when we call `bytesSkipped = (avail < remaining) ? avail : remaining`, `bytesSkipped` holds an int since this min statement bounds it to max of avail, which is an int.

I can attempt to do a bunch of refactoring if that will make things clearer but that shouldn't block this PR.